claudia-lab/test/fruit-test.js

```diff
@@ -0,0 +1,181 @@
+          expect(res.body.color).to.equal(examplefruit.color);
+          this.tempfruit = res.body;
+          done();
+        });
+      });
+    });
+
+    it ('should respond with 400 - bad request', done => {
+      request.post(`${url}/api/fruit/`)
+      .send({})
+      .end((err, res) => {
+        console.error('res status', res.status);
+        expect(res.status).to.equal(400);
+        expect(err).to.not.equal(null);
+        done();
+      });
+    });
+  });
+
+  describe('testing PUT requests to /api/fruit', function(){
+    before (done => { //before all tests are ran
+      //mocking the list for the test
+      Fruit.createFruit(examplefruit)
+      .then(fruit => {
+        this.tempfruit = fruit;
+        done();
+      })
+      .catch(err => done(err));
+    });
+
+    // remove mocked list after all tests have completed
+    after(done => {
+      Fruit.deleteFruit(this.tempfruit.id)
+      .then(() => done())
+      .catch(err => done(err));
+    });
+
+    describe('testing with a valid id and body', () => {
+
+      it ('should return a fruit', done => {
+        let updateContent = {name: 'new title', texture: 'new fruit', color: 'new date'};
+        request.put(`${url}/api/fruit/${this.tempfruit.id}`)
+        .send(updateContent) //send with updated content
+        .end((err,res) => {
+          if (err) return done(err);
+          console.log('response body', res.body);
+          console.log('temp fruit', this.tempfruit);
+          expect(res.status).to.equal(200);
+          expect(res.body.id).to.equal(this.tempfruit.id);
+          for (var key in updateContent) {
+            expect(res.body[key]).to.equal(updateContent[key]);
+          }
+          done();
+        });
+      });
+    });//end of testing valid id and body
+
+    it ('should respond with 400 - bad request', done => {
+      let updateContent = 'asdfasdf';
+      request.put(`${url}/api/fruit/${this.tempfruit.id}`)
+      .set('Content-Type', 'application/json')
+      .send(updateContent)
+      .end((err, res) => {
+        console.error('res status', res.status);
+        expect(res.status).to.equal(400);
+        expect(err).to.not.equal(null);
+        done();
+      });
+    });
+  });
+
+
+  describe('testing DELETE /api/fruit', function(){
+    before (done => { //before all tests are ran
+      //mocking the list for the test
+      Fruit.createFruit(examplefruit)
+      .then(fruit => {
+        this.tempfruit = fruit;
+        done();
+      })
+      .catch(err => done(err));
+    });
+
+    it ('should respond with 204 successful deletion', done => {
+      request.delete(`${url}/api/fruit/${this.tempfruit.id}`)
+      .end((err, res) => {
+        console.log(this.tempfruit.id);
+        expect(res.status).to.equal(204);
+        done();
+      });
+    });
+
+    it ('should respond with 404 not found', done => {
+      request.delete(`${url}/api/fruit/blerg`)
```

Comment:
@loomnugget not sure what's going on here - testing a specific test fruit maybe?